client/src/test/java/org/eclipse/hono/client/impl/HonoClientImplTest.java

```diff
@@ -311,6 +311,61 @@ public void testOnRemoteCloseTriggersReconnection(final TestContext ctx) {
         // THEN the client invokes the disconnect handler provided in the original connect method call
         disconnectHandlerInvocation.await(500);
     }
+    
+    /**
+     * Verifies that it fails to connect after client was shutdown.
+     * 
+     * @param ctx The test context.
+     *
+     */
+    @Test
+    public void testConnectFailsAfterShutdown(final TestContext ctx) {
+
+        // GIVEN a shutdown client
+        ConnectionFactory connectionFactory = mock(ConnectionFactory.class);
+        HonoClientImpl client = new HonoClientImpl(vertx, connectionFactory);
+        client.shutdown();
+        
+        // WHEN client gets connects 
+        Async connectionHandlerInvocation = ctx.async();
+        Handler<AsyncResult<HonoClient>> connectionHandler = result -> {
+            if(result.failed()) {
+                connectionHandlerInvocation.complete();
+            }
+        };
+        client.connect(new ProtonClientOptions(), connectionHandler );
+        
+        //THEN connect fails
+        connectionHandlerInvocation.await(500);
+    }
+    
+    /**
+     * Verifies that the client does not try to re-connect to a server instance if the client was shutdown.
+     * 
+     * @param ctx The test context.
+     *
+     */
+    @Test
+    public void testDoesNotTriggerReconnectionAfterShutdown(final TestContext ctx) {
+
+        // GIVEN a client that tries to connect to a server but does not succeed
+        ProtonConnection con = mock(ProtonConnection.class);
+        DisconnectHandlerProvidingConnectionFactory connectionFactory = new DisconnectHandlerProvidingConnectionFactory(con, Integer.MAX_VALUE, 1);
+        HonoClientImpl client = new HonoClientImpl(vertx, connectionFactory);
+        Async connectionHandlerInvocation = ctx.async();
+        Handler<AsyncResult<HonoClient>> connectionHandler = result -> {
+            if(result.failed()) {
+                connectionHandlerInvocation.complete();
+            }
+        };
+        client.connect(new ProtonClientOptions().setReconnectAttempts(1), connectionHandler);
```

Comment:
when this is executed, the connection attempt will fail once, then the `reconnect` will succeed and the connection handler will be completed.
The subsequent invocation of `shutdown` is not guaranteed to be executed *before* the (second) connection attempt ...

Reply:
Uh good catch, I switched arguments by accident. The Integer.MAX needs to be the second argument to just let connect always fail.